docx/text/tabstops.py

```diff
@@ -18,3 +18,16 @@ class TabStops(ElementProxy):
     using the `tab_stops` property of ParagraphFormat; it is not intended to
     be constructed directly.
     """
+
+    __slots__ = ()
+
+    def __len__(self):
+        tabs = self._get_tabs()
+        return len(tabs.tab_lst)
+
+    def _get_tabs(self):
+        """
+        Get or Create tabs list as needed.
+        """
+        tabs = self._element.get_or_add_tabs()
+        return tabs
```

Comment:
See the merged version for this. The general principle is that the API class (like TabStops) calls the oxml layer for whatever it needs in the XML domain. Most of that is provided by the meta class, so automatically available once you define the CT_x class and its ZeroOrOne's and whatever. However, if something fancier is needed, perhaps to navigate across more than one level in the XML hierarchy, additional methods are added in the oxml layer. So _get_tabs(), if it were needed, would go in CT_Tabs.

It's not needed in this case though, you can see the merged version for how that works out. Basically, the w:tabs element needs to have something in it (at least one w:tab element) or be removed. So we can't add one just to see how many are in it. We have to test for the not present condition and return zero in that case.